src/fenics_constitutive/solver.py

```diff
@@ -293,14 +295,21 @@ def form(self, x: PETSc.Vec) -> None:
 
             with df.common.Timer("stress_evaluation"):
                 self.stress_1.x.array[:] = self.stress_0.x.array
-                if law.history_dim is not None:
+                history_input = None
+                if isinstance(law.history_dim, int):
```

Comment:
This option was removed in the last PR. I am not sure why it wants to overwrite the stuff on main. Could you merge main into your branch first?